spec/services/encryption/encryptors/session_encryptor_spec.rb

```diff
@@ -0,0 +1,79 @@
+require 'rails_helper'
+
+describe Encryption::Encryptors::SessionEncryptor do
+  let(:plaintext) { '{ "foo": "bar" }' }
+
+  before do
+    described_class.instance_variable_set(:@user_access_key_scrypt_hash, nil)
+  end
+
+  describe '#encrypt' do
+    it 'returns encrypted text' do
+      ciphertext = subject.encrypt(plaintext)
+
+      expect(ciphertext).to_not eq(plaintext)
+    end
+
+    it 'only computes an scrypt hash on the first encryption' do
```

Comment:
This looks to be testing the behavior of `UserAccessKey` when `scrypt_hash` is provided on initialization. If so, perhaps this test belongs in the spec for that, and here all you really need to test is whether or not you provide `scrypt_hash` when initializing `UserAccessKey`. Then you're mocking the direct dependency of this class instead of one that is several steps away. This prevents a situation where some internal logic changes upstream of this and breaks this test, even though the api of that upstream class hasn't changed.

This follows for some other tests below as well.